flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergSink.java

```diff
@@ -0,0 +1,740 @@
+     * prepended with a prefix like "Sink Committer: uidSuffix".
+     *
+     * <p>Flink auto generates operator uid if not set explicitly. It is a recommended <a
+     * href="https://ci.apache.org/projects/flink/flink-docs-master/docs/ops/production_ready/">
+     * best-practice to set uid for all operators</a> before deploying to production. Flink has an
+     * option to {@code pipeline.auto-generate-uid=false} to disable auto-generation and force
+     * explicit setting of all operator uid.
+     *
+     * <p>Be careful with setting this for an existing job, because now we are changing the operator
+     * uid from an auto-generated one to this new value. When deploying the change with a
+     * checkpoint, Flink won't be able to restore the previous IcebergSink operator state (more
+     * specifically the committer operator state). You need to use {@code --allowNonRestoredState}
+     * to ignore the previous sink state. During restore IcebergSink state is used to check if last
+     * commit was actually successful or not. {@code --allowNonRestoredState} can lead to data loss
+     * if the Iceberg commit failed in the last completed checkpoint.
+     *
+     * @param newSuffix suffix for Flink sink operator uid and name
+     * @return {@link Builder} to connect the iceberg table.
+     */
+    public Builder uidSuffix(String newSuffix) {
+      this.uidSuffix = newSuffix;
+      return this;
+    }
+
+    public Builder snapshotProperties(Map<String, String> properties) {
+      snapshotSummary.putAll(properties);
+      return this;
+    }
+
+    public Builder setSnapshotProperty(String property, String value) {
+      snapshotSummary.put(property, value);
+      return this;
+    }
+
+    public Builder toBranch(String branch) {
+      writeOptions.put(FlinkWriteOptions.BRANCH.key(), branch);
+      return this;
+    }
+
+    private String operatorName(String suffix) {
+      return uidSuffix != null ? uidSuffix + "-" + suffix : suffix;
+    }
+
+    IcebergSink build() {
+
+      Preconditions.checkArgument(
+          inputCreator != null,
+          "Please use forRowData() or forMapperOutputType() to initialize the input DataStream.");
+      Preconditions.checkNotNull(tableLoader(), "Table loader shouldn't be null");
+
+      // Set the table if it is not yet set in the builder, so we can do the equalityId checks
+      this.table = checkAndGetTable(tableLoader(), table);
+
+      // Init the `flinkWriteConf` here, so we can do the checks
+      FlinkWriteConf flinkWriteConf = new FlinkWriteConf(table, writeOptions, readableConfig);
+
+      Duration tableRefreshInterval = flinkWriteConf.tableRefreshInterval();
+      SerializableSupplier<Table> tableSupplier;
+      if (tableRefreshInterval != null) {
+        tableSupplier = new CachingTableSupplier(table, tableLoader(), tableRefreshInterval);
+      } else {
+        SerializableTable serializableTable = (SerializableTable) SerializableTable.copyOf(table);
```

Comment:
this line is not needed since `table` is already a `SerializableTable`

Reply:
| what is the error/exception you are seeing after removed the 1st global?

TestIcebergSink failures for parallelism > 1:
```
java.lang.UnsupportedOperationException: Forward partitioning does not allow change of parallelism. Upstream operation: Sink: Writer-36 parallelism: 2, downstream operation: Sink: -hadoop.default.t-e254fac5-1ac5-40f1-ac6f-87cf37af0de9-pre-commit-topology-38 parallelism: 1 You must use another partitioning strategy, such as broadcast, rebalance, shuffle or global.
	at org.apache.flink.streaming.api.graph.StreamGraph.createActualEdge(StreamGraph.java:697)
	at org.apache.flink.streaming.api.graph.StreamGraph.addEdgeInternal(StreamGraph.java:649)
	at org.apache.flink.streaming.api.graph.StreamGraph.addEdgeInternal(StreamGraph.java:639)
	at org.apache.flink.streaming.api.graph.StreamGraph.addEdge(StreamGraph.java:596)
	at org.apache.flink.streaming.api.graph.StreamGraph.addEdge(StreamGraph.java:588)
```